pkg/agent/watch.go

```diff
@@ -123,3 +123,30 @@ func makeVMEvent(vm *vmapi.VirtualMachine, kind vmEventKind) (vmEvent, error) {
 		podIP:   vm.Status.PodIP,
 	}, nil
 }
+
+// custom formatting for vmEvent so that it prints in the same way as VmInfo
+func (e vmEvent) Format(state fmt.State, verb rune) {
+	switch {
+	case verb == 'v' && state.Flag('#'):
+		state.Write([]byte(fmt.Sprintf(
+			// note: intentionally order podName and podIP before vmInfo because vmInfo is large.
+			"agent.vmEvent{kind:%q, podName:%q, podIP:%q, vmInfo:%#v}",
+			e.kind, e.podName, e.podIP, e.vmInfo,
+		)))
+	default:
+		if verb != 'v' {
+			state.Write([]byte("%!"))
+			state.Write([]byte(string(verb)))
+			state.Write([]byte("(api.VmInfo="))
```

Comment:
why do you sprintf and call Write once in other cases?

Reply:
Honestly, only because `api.VmInfo` does it that way, and to match that. Happy to change if you think `sprintf` would be cleaner

https://github.com/neondatabase/autoscaling/blob/2f2b73eb8f8802c7a6be9012b23e3c171b1a8a60/pkg/api/vminfo.go#L298-L302